### PR TITLE
[CoreBundle] Modernizr: alert hidable

### PR DESCRIPTION
**Pull request by ngodfraind**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Fixed tickets | #1422

### CODE REVIEW

**Comment by ptsavdar** [comment]
LGTM

File: main/core/Manager/PluginManager.php [hunk 1]
```diff
@@ -381,4 +384,48 @@ private function getPluginShortName($plugin)
 
         return count($parts) === 3 ? $parts[2] : $name;
     }
+
+    public function getCurrentDistributionCommit()
```
Comment (stefk) [inline main/core/Manager/PluginManager.php]: This has nothing to do with the plugin manager. Plus there are methods on the composer repository class which allow to get information in a few lines only, so you might as well put it in the twig extension directly.

File: main/core/Resources/config/routing.yml [hunk 2]
```diff
@@ -9,6 +9,11 @@ claro_oauth_authorize:
     resource: "@ClarolineCoreBundle/Controller/OauthController.php"
     type: annotation
 
+claro_layout:
```
Comment (stefk) [inline main/core/Resources/config/routing.yml]: Probably not needed anymore.

File: main/core/Twig/PluginExtension.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        return 'plugin_extension';
+    }
+
+    public function getCurrentDistributionCommit()
+    {
+        return $this->pluginManager->getCurrentDistributionCommit();
+    }
+
+    public function getCurrentUrl()
```
Comment (stefk) [inline main/core/Twig/PluginExtension.php]: ?

File: main/core/Twig/ComposerExtension.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    public function getName()
+    {
+        return 'plugin_extension';
+    }
+
+    public function getCurrentDistributionCommit()
+    {
+        $previous = $this->openRepository($this->installedRepoFile);
+
+        foreach ($previous->getCanonicalPackages() as $package) {
```
Comment (stefk) [inline main/core/Twig/ComposerExtension.php]: https://github.com/composer/composer/blob/master/src/Composer/Repository/RepositoryInterface.php#L46

File: main/core/Twig/ComposerExtension.php [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    public function getName()
+    {
+        return 'plugin_extension';
+    }
+
+    public function getCurrentDistributionCommit()
+    {
+        $previous = $this->openRepository($this->installedRepoFile);
```
Comment (stefk) [inline main/core/Twig/ComposerExtension.php]: Not previous.

File: main/core/Twig/ComposerExtension.php [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+    /**
+     * @param string $repoFile
+     * @param bool   $filter
+     *
+     * Currently copy pasted from OperationExecutor
+     *
+     * @return InstalledFilesystemRepository
+     */
+    private function openRepository($repoFile, $filter = true)
```
Comment (stefk) [inline main/core/Twig/ComposerExtension.php]: This method is mostly useless as you don't test or use filters. Could be replaced with:
```php
$repo = new InstalledFilesystemRepository(new JsonFile($this->installedRepoFile));
```

File: main/core/Twig/ComposerExtension.php (hunk 4 above)
Comment (stefk) [inline main/core/Twig/ComposerExtension.php]: Or https://github.com/composer/composer/blob/master/src/Composer/Repository/RepositoryInterface.php#L73

**Comment by stefk** [comment]
LGTM

**Pull request merged**